ios/Codeforces Watcher/Application/features/contests/ContestsNavigationBar.swift

```diff
@@ -0,0 +1,121 @@
+import SwiftUI
+
+struct ContestsNavigationBar: View {
+    
+    var filterItems: [ContestFilterView.UIModel]
+    
+    private let filtersViewAnimationDuration = 0.3
+    @State private var isContestFiltersView = false
+    @State private var geometryHeight: CGFloat = 0
+    
+    var body: some View {
+        HStack {
+            CommonText((isContestFiltersView ? "filters" : "Contests").localized)
+                .font(.headerMedium)
+                .foregroundColor(Palette.black.swiftUIColor)
+                .animation(nil)
+                .modifier(Shake(animatableData: isContestFiltersView ? 1 : 0))
+            
+            Spacer()
+            
+            if isContestFiltersView {
```

Comment:
It looks like both cases are almost identical aside from the actual image name. WDYT about extracting this view to private component and passing image name as parameter?

Reply:
And then `filtersViewAnimationDuration` will be private to this component as an implementation detail.